cachi2/core/package_managers/gomod.py

```diff
@@ -619,19 +619,18 @@ def fetch_gomod_source(request: Request) -> RequestOutput:
             components.extend(module.to_component() for module in modules)
             components.extend(package.to_component() for package in packages)
 
-        if "gomod-vendor-check" not in request.flags and "gomod-vendor" not in request.flags:
```

Comment:
If this check was enough previously, why can't we just do:
```
if app_dir.join_within_root("vendor").path.is_dir():
   ...
```

to retain the same logic. That would work, wouldn't it? Instead we're pushing it down to `_resolve_gomod` and fiddling with the GOMODCACHE setting which to me is unnecessary if we simply don't copy the temporary cache to the output and just leave it be collected by the tempdir context manager.

Reply:
The problem is that `app_dir` (or `main_module_dir` as it is called in this scope) only exists in the context of each `subpath` being processed. We can have a subpath that contains a `vendor` dir, and another one that doesn't ><

This check used to be request-global, and now we need to do it for each input package.

Reply:
We still want to copy the contents of `pkg/mod/cache/download` to the global output dir, considering that it will be populated by all input packages/subpaths that did not have a `vendor` dir.

Reply:
Thanks for the clarification.